providers/packetframe/packetframeProvider.go

```diff
@@ -0,0 +1,237 @@
+}
+
+// GetNameservers returns the nameservers for a domain.
+func (api *packetframeProvider) GetNameservers(domain string) ([]*models.Nameserver, error) {
+	return models.ToNameservers(defaultNameServerNames)
+}
+
+// GetZoneRecords gets the records of a zone and returns them in RecordConfig format.
+func (api *packetframeProvider) GetZoneRecords(domain string) (models.Records, error) {
+
+	if api.domainIndex == nil {
+		if err := api.fetchDomainList(); err != nil {
+			return nil, err
+		}
+	}
+	zone, ok := api.domainIndex[domain+"."]
+	if !ok {
+		return nil, fmt.Errorf("'%s' not a zone in Packetframe account", domain)
+	}
+
+	records, err := api.getRecords(zone.ID)
+	if err != nil {
+		return nil, fmt.Errorf("could not load records for '%s'", domain)
+	}
+
+	existingRecords := make([]*models.RecordConfig, len(records))
+
+	dc := models.DomainConfig{
+		Name: domain,
+	}
+
+	for i := range records {
+		existingRecords[i] = toRc(&dc, &records[i])
+	}
+
+	return existingRecords, nil
+}
+
+// GetDomainCorrections returns the corrections for a domain.
+func (api *packetframeProvider) GetDomainCorrections(dc *models.DomainConfig) ([]*models.Correction, error) {
+	dc, err := dc.Copy()
+	if err != nil {
+		return nil, err
+	}
+
+	dc.Punycode()
+
+	if api.domainIndex == nil {
+		if err := api.fetchDomainList(); err != nil {
+			return nil, err
+		}
+	}
+	zone, ok := api.domainIndex[dc.Name+"."]
+	if !ok {
+		return nil, fmt.Errorf("'%s' not a zone in Packetframe account", dc.Name)
+	}
+
+	records, err := api.getRecords(zone.ID)
+	if err != nil {
+		return nil, fmt.Errorf("could not load records for '%s'", dc.Name)
+	}
+
+	existingRecords := make([]*models.RecordConfig, len(records))
+
+	for i := range records {
+		existingRecords[i] = toRc(dc, &records[i])
+	}
+
+	// Normalize
+	models.PostProcessRecords(existingRecords)
+
+	differ := diff.New(dc)
+	_, create, delete, modify, err := differ.IncrementalDiff(existingRecords)
+	if err != nil {
+		return nil, err
+	}
+
+	var corrections []*models.Correction
+
+	for _, m := range create {
+		req, err := toReq(zone.ID, dc, m.Desired)
+		if err != nil {
+			return nil, err
+		}
+		corr := &models.Correction{
+			Msg: m.String(),
+			F: func() error {
+				_, err := api.createRecord(req)
+				return err
+			},
+		}
+		corrections = append(corrections, corr)
+	}
+
+	for _, m := range delete {
+		original := m.Existing.Original.(*domainRecord)
+		corr := &models.Correction{
+			Msg: fmt.Sprintf("Deleting record %s from %s", original.ID, zone.Zone),
+			F: func() error {
+				err := api.deleteRecord(zone.ID, original.ID)
+				return err
+			},
+		}
+		corrections = append(corrections, corr)
+	}
+
+	for _, m := range modify {
+		original := m.Existing.Original.(*domainRecord)
+		req, _ := toReq(zone.ID, dc, m.Desired)
+		req.ID = original.ID
+		corr := &models.Correction{
+			Msg: fmt.Sprintf("Modifying record %s from %s", original.ID, zone.Zone),
```

Comment:
Suggestion:

    Msg: fmt.Sprintf("Modifying record %q from %q", original.ID, zone.Zone),

(I'm assuming that ID is a string. If it isn't, use `'%s'`